airflow/models/dagbag.py

```diff
@@ -408,13 +405,11 @@ def collect_dags(
         self.dagbag_stats = sorted(
             stats, key=lambda x: x.duration, reverse=True)
         for file_stat in self.dagbag_stats:
-            dag_ids = dags_by_name[file_stat.dags]
-            if file_stat.dag_num >= 1:
-                # if we found multiple dags per file, the stat is 'dag_id1 _ dag_id2'
-                dag_names = '_'.join(dag_ids)
-                Stats.timing('dag.loading-duration.{}'.
-                             format(dag_names),
-                             file_stat.duration)
+            # file_stat.file similar format: /subdir/dag_name.py
+            filename = file_stat.file.split('/')[-1].replace('.py', '')
```

Comment:
it's possible to have the same file name in different folders. please can we join the split, instead of just taking the filename? eg.
```
filename = '.'.join(file_stat.file.split('/')).replace('.py', '')
```

Reply:
i was imagining a file structure like the for example. it doesn't seem too unreasonable a structure?
```
dag_dir
   +->dag_module_1
   |       +->dag_definition.py
   |       |->callbacks.py
   |       |->functions.py
   |       |->config.json
   +->dag_module_2
           +->dag_definition.py
           |->callbacks.py
           |->functions.py
           |->config.json
```

Reply:
but yea, we dont use that dir structure, so im happy with the merge :)